### DISCUSSION TITLE
title表格使用efcore注入数据服务，弹窗无编辑后关闭报tracked错误。

### DISCUSSION THREAD

**Discussion by KasimYe** [dotnetcore/BootstrapBlazor#4068] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

点击表格行中的编辑按钮，弹窗不执行修改操作直接关闭，报tracked错误。

### Steps To Reproduce

复现操作截图如下：
![image](https://github.com/user-attachments/assets/237ad1db-1f54-4aee-b32c-a466661007b6)
![image](https://github.com/user-attachments/assets/5b981614-37a2-41c0-85f8-509004c4c9a9)
复现代码如下：
https://github.com/KasimYe/PrintSerEFCoreTest

**Comment by bb-auto** [comment]
@KasimYe Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Reply by ArgoZhang** [reply, marked as answer]
@KasimYe 由于组件未开启 `IsTracking` 编辑时内部克隆了一个实体类进行编辑，`EFCore` 数据服务取消时，不需要进行 `CancelAsync` 即可

**Comment by HongFei-Ye** [comment]
[PrintSerEFCoreTest-main.zip](https://github.com/user-attachments/files/16621350/PrintSerEFCoreTest-main.zip)

增加检查是否已经跟踪了具有相同主键的实体，如果找到了已经被跟踪的相同主键实体，将其分离。

**Comment by sundows** [comment]
在 @NiuBiA 大佬的修改方案基础上，再完善一下，DefaultDataService 类的中 DeleteAsync 方法也要做同样分离的操作，不然也会发生 tracked 的错误。
![Y (60%SN0S656G4QMY@_0(Y](https://github.com/user-attachments/assets/aa4cbbcc-5314-453c-a0f8-bc6d91b9964e)
